frontend/src/core/codemirror/markdown/commands.ts

```diff
@@ -274,11 +277,55 @@ export async function insertImage(view: EditorView, file: File) {
     reader.readAsDataURL(file);
   });
 
+  let savedFilePath: string | undefined;
+
+  // If the file is base64 encoded, we can save it locally to prevent large file strings
+  try {
+    if (dataUrl.startsWith("data:")) {
+      const base64 = dataUrl.split(",")[1];
+      const inputFilename = prompt(
+        "We can save your image as a file. Please enter a filename.",
+      );
+
+      if (inputFilename) {
+        const extension = file.type.split("/")[1];
+        const filename = `${inputFilename}.${extension}`;
```

Comment:
Super small QOL but could check if it ends with the extension already before adding